woocommerce/woocommerce
# Add/32 react provider to form component

### All Submissions:






### Changes proposed in this Pull Request:

This adds the use of a React context provider to the Form component. This will allow nested fields to more easily make use of the Form data without having to property drill down.

Closes https://github.com/woocommerce/mothra-private/issues/32

### How to test the changes in this Pull Request:

1. Load this PR and start the onboarding wizard on a fresh WooCommerce site
2. Step through the onboarding wizard and make sure the Store details and Business detail forms work correctly still
3. Check that the tests work correctly
4. You can create your own component as per the [README](https://github.com/woocommerce/woocommerce/tree/add/32_react_provider_to_form_component/packages/js/components/src/form#usage-with-formcontext) and see if things work as expected.

### Other information:




### FOR PR REVIEWER ONLY:
